### PR TITLE
Add Spanish translation of serialization.mdx

### PR DESCRIPTION
**Pull request by donelianc**
# What does this PR do?

Add the Spanish translation for `serialization.mdx` as part of the #15947 issue.
Changes include the Spanish version of the original document and the updated `_toctree.yml` file.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests) Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://github.com/huggingface/transformers/issues/15947#issuecomment-1312741421)**.
- [x] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by donelianc** [comment]
@omarespejel or @osanseviero, can you help me review this PR? Thanks!

**Review by sgugger — APPROVED** [review]
LGTM in general, but I'll let @osanseviero proofread :-)

File: docs/source/es/serialization.mdx [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+
+# Exportar modelos 🤗 Transformers
+
+Si necesitas implementar modelos 🤗 Transformers en entornos de producción, te 
+recomendamos exportarlos a un formato serializado que se pueda cargar y ejecutar 
+en tiempos de ejecución y hardware especializados. En esta guía, te mostraremos cómo 
+exportar modelos 🤗 Transformers en dos formatos ampliamente utilizados: ONNX y TorchScript.
+
+Una vez exportado, un modelo puede optimizarse para la inferencia a través de técnicas 
+como la cuantificación y _pruning_. Si estás interesado en optimizar tus modelos para
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:21]:
```suggestion
como la cuantización y _pruning_. Si estás interesado en optimizar tus modelos para 
```

File: docs/source/es/serialization.mdx [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+[biblioteca de 🤗 Optimum](https://github.com/huggingface/optimum).
+
+## ONNX
+
+El proyecto [ONNX (Open Neural Network eXchange)](http://onnx.ai) es un 
+estándar abierto que define un conjunto común de operadores y un formato 
+de archivo común para representar modelos de aprendizaje profundo en una 
+amplia variedad de _frameworks_, incluidos PyTorch y TensorFlow. Cuando un modelo 
+se exporta al formato ONNX, estos operadores se usan para construir un 
+gráfico computacional (a menudo llamado _representación intermedia_) que
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:32]:
```suggestion
grafo computacional (a menudo llamado _representación intermedia_) que 
```

File: docs/source/es/serialization.mdx [hunk 3]
```diff
@@ -0,0 +1,669 @@
...
+
+El proyecto [ONNX (Open Neural Network eXchange)](http://onnx.ai) es un 
+estándar abierto que define un conjunto común de operadores y un formato 
+de archivo común para representar modelos de aprendizaje profundo en una 
+amplia variedad de _frameworks_, incluidos PyTorch y TensorFlow. Cuando un modelo 
+se exporta al formato ONNX, estos operadores se usan para construir un 
+gráfico computacional (a menudo llamado _representación intermedia_) que 
+representa el flujo de datos a través de la red neuronal.
+
+Al exponer un gráfico con operadores y tipos de datos estandarizados, ONNX facilita
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:35]:
```suggestion
Al exponer un grafo con operadores y tipos de datos estandarizados, ONNX facilita 
```

File: docs/source/es/serialization.mdx [hunk 4]
```diff
@@ -0,0 +1,669 @@
...
+se exporta al formato ONNX, estos operadores se usan para construir un 
+gráfico computacional (a menudo llamado _representación intermedia_) que 
+representa el flujo de datos a través de la red neuronal.
+
+Al exponer un gráfico con operadores y tipos de datos estandarizados, ONNX facilita 
+el cambio entre frameworks. Por ejemplo, un modelo entrenado en PyTorch se puede 
+exportar a formato ONNX y luego importar en TensorFlow (y viceversa).
+
+🤗 Transformers proporciona un paquete llamado `transformers.onnx`, el cual permite convertir 
+puntos de control de un modelo en un gráfico ONNX aprovechando los objetos de configuración.
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:40]:
```suggestion
puntos de control de un modelo en un grafo ONNX aprovechando los objetos de configuración. 
```

File: docs/source/es/serialization.mdx [hunk 5]
```diff
@@ -0,0 +1,669 @@
...
+- YOLOS
+
+En las próximas dos secciones, te mostraremos cómo:
+
+* Exportar un modelo compatible utilizando el paquete `transformers.onnx`.
+* Exportar un modelo personalizado para una arquitectura no compatible.
+
+### Exportar un model a ONNX
+
+Para exportar un modelo 🤗 Transformers a ONNX, tienes que instalar primer algunas
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:105]:
```suggestion
Para exportar un modelo 🤗 Transformers a ONNX, tienes que instalar primero algunas
```

File: docs/source/es/serialization.mdx [hunk 6]
```diff
@@ -0,0 +1,669 @@
...
+  -h, --help            show this help message and exit
+  -m MODEL, --model MODEL
+                        Model ID on huggingface.co or path on disk to load model from.
+  --feature {causal-lm, ...}
+                        The type of features to export the model with.
+  --opset OPSET         ONNX opset version to export the model with.
+  --atol ATOL           Absolute difference tolerence when validating the model.
+```
+
+Exportar un punto de control usando una configuración a la medida se puede hacer de la siguiente manera:
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:132]: Punto de control is not used afaik 
```suggestion
Exportar un checkpoint usando una configuración a la medida se puede hacer de la siguiente manera:
```

File: docs/source/es/serialization.mdx [hunk 7]
```diff
@@ -0,0 +1,669 @@
...
+        -[✓] ONNX model output names match reference model ({'last_hidden_state'})
+        - Validating ONNX Model output "last_hidden_state":
+                -[✓] (2, 8, 768) matches (2, 8, 768)
+                -[✓] all values close (atol: 1e-05)
+All good, model saved at: onnx/model.onnx
+```
+
+Esto exporta un gráfico ONNX del punto de control definido por el argumento `--model`. 
+En este ejemplo, es un modelo `distilbert-base-uncased`, pero puede ser cualquier punto 
+de control en Hugging Face Hub o que esté almacenado localmente.
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:151]:
```suggestion
En este ejemplo, es un modelo `distilbert-base-uncased`, pero puede ser cualquier checkpoint en Hugging Face Hub o que esté almacenado localmente.
```

File: docs/source/es/serialization.mdx [hunk 8]
```diff
@@ -0,0 +1,669 @@
...
+```python
+>>> from transformers.models.distilbert import DistilBertConfig, DistilBertOnnxConfig
+
+>>> config = DistilBertConfig()
+>>> onnx_config = DistilBertOnnxConfig(config)
+>>> print(list(onnx_config.outputs.keys()))
+["last_hidden_state"]s
+```
+
+El proceso es idéntico para los puntos de control de TensorFlow en Hub.
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:181]:
```suggestion
El proceso es idéntico para los checlpoints de TensorFlow en Hub. 
```

File: docs/source/es/serialization.mdx [hunk 9]
```diff
@@ -0,0 +1,669 @@
...
+>>> from transformers.models.distilbert import DistilBertConfig, DistilBertOnnxConfig
+
+>>> config = DistilBertConfig()
+>>> onnx_config = DistilBertOnnxConfig(config)
+>>> print(list(onnx_config.outputs.keys()))
+["last_hidden_state"]s
+```
+
+El proceso es idéntico para los puntos de control de TensorFlow en Hub. 
+Por ejemplo, podemos exportar un punto de control puro de TensorFlow desde
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:182]:
```suggestion
Por ejemplo, podemos exportar un checkpoint puro de TensorFlow desde 
```

File: docs/source/es/serialization.mdx [hunk 10]
```diff
@@ -0,0 +1,669 @@
...
+Por ejemplo, podemos exportar un punto de control puro de TensorFlow desde 
+[Keras](https://huggingface.co/keras-io) de la siguiente manera:
+
+```bash
+python -m transformers.onnx --model=keras-io/transformers-qa onnx/
+```
+
+Para exportar un modelo que está almacenado localmente, deberás tener los pesos 
+y tokenizadores del modelo almacenados en un directorio. Por ejemplo, podemos cargar 
+y guardar un punto de control de la siguiente manera:
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:191]:
```suggestion
y guardar un checkpoint de la siguiente manera:
```

File: docs/source/es/serialization.mdx [hunk 11]
```diff
@@ -0,0 +1,669 @@
...
+
+>>> # Load tokenizer and PyTorch weights form the Hub
+>>> tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased")
+>>> pt_model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased")
+>>> # Save to disk
+>>> tokenizer.save_pretrained("local-pt-checkpoint")
+>>> pt_model.save_pretrained("local-pt-checkpoint")
+```
+
+Una vez que se guarda el punto de control, podemos exportarlo a ONNX usando el argumento `--model`
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:206]: Here and everywhere else let's use `checkpoint` :)

File: docs/source/es/serialization.mdx [hunk 12]
```diff
@@ -0,0 +1,669 @@
...
+* Modelos _Encoder-decoder_ inherente de [`~onnx.config.OnnxSeq2SeqConfigWithPast`]
+
+<Tip>
+
+Una buena manera de implementar una configuración personalizada en ONNX es observar la implementación 
+existente en el archivo `configuration_<model_name>.py` de una arquitectura similar.
+
+</Tip>
+
+Dado que DistilBERT es un modelo de tipo _encoder-decoder_, su configuración se hereda de `OnnxConfig`:
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:321]: DistilBERT is an encoder model

```suggestion
Dado que DistilBERT es un modelo de tipo _encoder_, su configuración se hereda de `OnnxConfig`:
```

File: docs/source/es/serialization.mdx [hunk 13]
```diff
@@ -0,0 +1,669 @@
...
+...         return OrderedDict(
+...             [
+...                 ("input_ids", {0: "batch", 1: "sequence"}),
+...                 ("attention_mask", {0: "batch", 1: "sequence"}),
+...             ]
+...         )
+```
+
+Cada objeto de configuración debe implementar la propiedad `inputs` y devolver un mapeo, 
+donde cada clave corresponde a una entrada esperada y cada valor indica el eje de esa entrada.
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:340]:
```suggestion
donde cada llave corresponde a una entrada esperada y cada valor indica el eje de esa entrada. 
```

File: docs/source/es/serialization.mdx [hunk 14]
```diff
@@ -0,0 +1,669 @@
...
+donde cada clave corresponde a una entrada esperada y cada valor indica el eje de esa entrada. 
+Para DistilBERT, podemos ver que se requieren dos entradas: `input_ids` y `attention_mask`. 
+Estas entradas tienen la misma forma de `(batch_size, sequence_length)`, es por lo que vemos 
+los mismos ejes utilizados en la configuración.
+
+<Tip>
+
+Observa que la propiedad `inputs` para `DistilBertOnnxConfig` devuelve un `OrderedDict`.
+Esto nos asegura que las entradas coincidan con su posición relativa dentro del método 
+`PreTrainedModel.forward()` al rastrear el gráfico. Recomendamos usar un `OrderedDict`
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:349]:
```suggestion
`PreTrainedModel.forward()` al rastrear el grafo. Recomendamos usar un `OrderedDict` 
```

File: docs/source/es/serialization.mdx [hunk 15]
```diff
@@ -0,0 +1,669 @@
...
+
+Revisa cómo fue la contribución para la [configuración de IBERT](https://github.com/huggingface/transformers/pull/14868/files) 
+y así tener una idea de lo que necesito.
+
+## TorchScript
+
+<Tip>
+
+Este es el comienzo de nuestros experimentos con TorchScript y todavía estamos explorando sus capacidades con modelos de 
+tamaño de entrada variable. Es un tema de interés para nosotros y profundizaremos nuestro análisis en las próximas
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:481]:
```suggestion
tamaño de entrada variable. Es un tema de interés y profundizaremos nuestro análisis en las próximas 
```

File: docs/source/es/serialization.mdx [hunk 16]
```diff
@@ -0,0 +1,669 @@
...
+Revisa cómo fue la contribución para la [configuración de IBERT](https://github.com/huggingface/transformers/pull/14868/files) 
+y así tener una idea de lo que necesito.
+
+## TorchScript
+
+<Tip>
+
+Este es el comienzo de nuestros experimentos con TorchScript y todavía estamos explorando sus capacidades con modelos de 
+tamaño de entrada variable. Es un tema de interés para nosotros y profundizaremos nuestro análisis en las próximas 
+versiones,  con más ejemplos de código, una implementación más  flexible y puntos de referencia que comparen códigos
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:482]:
```suggestion
versiones,  con más ejemplos de código, una implementación más flexible y puntos de referencia que comparen códigos 
```

File: docs/source/es/serialization.mdx [hunk 17]
```diff
@@ -0,0 +1,669 @@
...
+basados en Python con TorchScript compilado.
+
+</Tip>
+
+Según la documentación de Pytorch: "TorchScript es una forma de crear modelos serializables y optimizables a partir del 
+código de PyTorch". Los dos módulos de Pytorch [JIT y TRACE](https://pytorch.org/docs/stable/jit.html) permiten al 
+desarrollador exportar su modelo para reutilizarlo  en otros programas, como los programas C++ orientados a la eficiencia.
+
+Hemos proporcionado una interfaz que permite exportar modelos de 🤗 Transformers a TorchScript para que puedan reutilizarse 
+en un entorno diferente  al de un programa Python basado en Pytorch. Aquí explicamos cómo exportar y usar nuestros modelos
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:492]:
```suggestion
en un entorno diferente al de un programa Python basado en Pytorch. Aquí explicamos cómo exportar y usar nuestros modelos 
```

File: docs/source/es/serialization.mdx [hunk 18]
```diff
@@ -0,0 +1,669 @@
...
+<Tip>
+
+Este es el comienzo de nuestros experimentos con TorchScript y todavía estamos explorando sus capacidades con modelos de 
+tamaño de entrada variable. Es un tema de interés para nosotros y profundizaremos nuestro análisis en las próximas 
+versiones,  con más ejemplos de código, una implementación más  flexible y puntos de referencia que comparen códigos 
+basados en Python con TorchScript compilado.
+
+</Tip>
+
+Según la documentación de Pytorch: "TorchScript es una forma de crear modelos serializables y optimizables a partir del
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:487]:
```suggestion
Según la documentación de PyTorch: "TorchScript es una forma de crear modelos serializables y optimizables a partir del 
```

Pytorch -> PyTorch here and below

File: docs/source/es/serialization.mdx [hunk 19]
```diff
@@ -0,0 +1,669 @@
...
+#### Dependencias
+
+Usar AWS Neuron para convertir modelos requiere las siguientes dependencias y entornos:
+
+* Un [entorno Neuron SDK](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-guide/neuron-frameworks/pytorch-neuron/index.html#installation-guide), 
+que viene preconfigurado en [AWS Deep Learning AMI](https://docs.aws.amazon.com/dlami/latest/devguide/tutorial-inferentia-launching.html).
+
+#### Convertir un modelo a AWS Neuron
+
+Con el mismo script usado en [Uso de TorchScript en Python](https://huggingface.co/docs/transformers/main/es/serialization#using-torchscript-in-python) ç
```
Comment (osanseviero) [inline docs/source/es/serialization.mdx:645]:
```suggestion
Con el mismo script usado en [Uso de TorchScript en Python](https://huggingface.co/docs/transformers/main/es/serialization#using-torchscript-in-python)
```

**Review by osanseviero — APPROVED** [review]
Very cool contribution! Muchas gracias 🔥 

I left a couple of comments, but my main request is to not translate checkpoints and to translate graph as grafo and not graficos. Thanks again!

File: docs/source/es/serialization.mdx (hunk 11 above)
Comment (donelianc) [reply docs/source/es/serialization.mdx]: I'll consider this for future translations, thx!

**Comment by donelianc** [comment]
@osanseviero, I addressed your comments in my last commit. Thanks for your review. I'll consider your feedback for future translations.

**Review by osanseviero — APPROVED** [review]
Thanks a lot! 🔥 great work!

**Comment by donelianc** [comment]
@sgugger, review done :) Please merge when possible. Thanks!

**Review by sgugger — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**